core/src/main/scala/scalafix/util/TreeExtractors.scala

```diff
@@ -0,0 +1,8 @@
+package scalafix.util
+import scala.meta._
+object TreeExtractors {
+  object `:WithParent:` {
+    def unapply(tree: Tree): Option[(Tree, Tree)] =
+      tree.parent.map(parent => tree -> parent)
+  }
+}
```

Comment:
I was using it, and I have a similar one in scalafmt. Should probably be in contrib.